tests/integration/test_zfs_generic.py

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-module-docstring
+# pylint: disable=missing-function-docstring
+
+import pytest
+
+from tests.integration.infra import repl_invoke, dump_exists, slurp_output_file
+
+CMD_TABLE = [
```

Comment:
I'm good with this as is, but after seeing the duplication in the test files, I wonder if there might be a more elegant way..

e.g. perhaps we could have a single test file, and combine all commands into a single list in that file.. or split the commands into plain text files (e.g. one command per line), and read those files and feed the commands into pytest..

I'm good with what you have, just something I thought about as I looked this over.

Reply:
I did think about that. The main reason really why I didn't do that is that having them in string form within Python seems simpler/more straightforward to me than writing code to read files and construct them in the runtime (which is not very complicated - just less simple than having them in string form. I'd be open to making this in a future or in this change if you think we have reasons to do so.